pkg/api/controller.go

```diff
@@ -1108,13 +1108,13 @@ func (c *Controller) CreateRepository(w http.ResponseWriter, r *http.Request, bo
 		return
 	}
 
-	err := ensureStorageNamespaceRW(ctx, c.BlockAdapter, body.StorageNamespace)
+	err := ensureStorageNamespace(ctx, c.BlockAdapter, body.StorageNamespace)
 	if err != nil {
 		c.Logger.
 			WithError(err).
 			WithField("storage_namespace", body.StorageNamespace).
 			Warn("Could not access storage namespace")
-		writeError(w, http.StatusBadRequest, "error creating repository: could not access storage namespace")
+		writeError(w, http.StatusBadRequest, "error creating repository: "+err.Error())
```

Comment:
Yikes... this can leak potentially sensitive information.  A good way to handle something like this is to return _two_ values (and document them!) from `ensureStorageNamespace`: the first is an `error`, the second is a `string` which will be shown to the user.  Then it is easier to ensure that nobody ever returns sensitive information from that function.